orderer/consensus/smartbft/util_network_test.go

```diff
@@ -233,14 +242,14 @@ func (n *Node) sendRequest(sender uint64, target uint64, request []byte) error {
 }
 
 func (n *Node) receiveMessage(sender uint64, message *smartbftprotos.Message) {
-	n.lock.Lock()
-	defer n.lock.Unlock()
+	n.lock.RLock()
+	defer n.lock.RUnlock()
```

Comment:
why was this changed from Lock() to RLock()?

I am not sure it is related but:
It seems to me that the main problem here is that the routine that does `send` also invokes the `receive` and the `Handle`. In a real network that does not happen, there is a go-routine and the receiving end that reads incoming messages and invokes the "Handle".